R/iregnet.R

```diff
@@ -189,11 +189,11 @@ iregnet <- function(x, y,
     status <- get_status_from_surv(y)
     y <- as.matrix(y[, 1:(ncol(y)-1)])
   } else {
-    stopifnot_error("y should be a 2 column matrix", ncol(y) == 2)
-    # Check if completely left or right censored
-    check_censorship(y)
+    y <- as.matrix(y)
```

Comment:
Does this work okay with `Surv` objects as inputs?

Reply:
That line is in an else block which will be executed when the target isn't `surv` object, if it is a `surv` object then `get_status_from_surv` will handle it. 
```suggestion
   if (survival::is.Surv(y)) {
    status <- get_status_from_surv(y)
    y <- as.matrix(y[, 1:(ncol(y)-1)])
  } else {
    y <- as.matrix(y)
    stopifnot_error("y should be a 2 column matrix", ncol(y) <= 2)
    if(ncol(y) == 1)
      y <- cbind(y, y)
    else check_censorship(y) # Check if completely left or right censored
  }
```